lib/flipper/gates/percentage_of_actors.rb

```diff
@@ -26,14 +26,16 @@ def enabled?(value)
       SCALING_FACTOR = 1_000
       private_constant :SCALING_FACTOR
 
-      # Internal: Checks if the gate is open for a thing.
+      # Internal: Checks if the gate is open for one or more actors.
       #
-      # Returns true if gate open for thing, false if not.
+      # Returns true if gate open for any actors, false if not.
       def open?(context)
-        return false if context.thing.nil?
+        return false unless context.actors?
 
-        id = "#{context.feature_name}#{context.thing.value}"
-        Zlib.crc32(id) % (100 * SCALING_FACTOR) < context.values.percentage_of_actors * SCALING_FACTOR
+        context.actors.any? do |actor|
+          id = "#{context.feature_name}#{actor.value}"
+          Zlib.crc32(id) % (100 * SCALING_FACTOR) < context.values.percentage_of_actors * SCALING_FACTOR
+        end
```

Comment:
I think this will change the probability. For example, if this is enabled for 50% and there are 6 actors, it would be the same as rolling a dice 6 times and seeing if **any** of the dice rolls are < 4. Each additional actor increases the probability (6x in this example).

More accurately would be to calculate the CRC once and compare that to a percentage:

```ruby
id = "#{context.feature_name}#{context.actors.map(&:value).join(';')}"
Zlib.crc32(id) % (100 * SCALING_FACTOR) < context.values.percentage_of_actors * SCALING_FACTOR
```

Reply:
Hmm. I’m not sure what is right. I was trying to make it the same probability as checking each actor individually would be. 

If we change this then how will people migrate to get the array of actors benefit?

Reply:
Resolved by 63f4f695253ded4b138825f808c754257eac5a1d